proposals/0249-commission-update-rules.md

```diff
@@ -0,0 +1,135 @@
+---
+simd: '0249'
+title: Commission Update Rules
+authors: Justin Starry (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-02-18
+feature: (fill in with feature key and github tracking issues once accepted)
+---
+
+## Summary
+
+Allow validators to update their inflations reward commission rate with basis
+points at any time but delay those commission updates for at least one full
+epoch. 
+
+## Motivation
+
+Validators should have more fine grained control over commission values and
+should be able update their desired commission at any time. That said, stake
+delegators should always have ample time to re-delegate their stake in response
+to validator commission rate changes.
+
+## New Terminology
+
+NA
+
+## Detailed Design
+
+### Runtime
+
+During inflation reward distribution for an epoch `E`, use the inflation rewards
+commission rate set in the vote account state at the beginning of the epoch `E -
+1` when calculating validator commissions. This is the same vote account state
+used to calculate the leader schedule for epoch `E` so it must already be
+available in-protocol. The only exception is inflation reward distribution for
+epoch `E == 0`. In that case, use the commission rate set by vote accounts in
+the genesis config.
```

Comment:
How exactly does this work for new validators?

If I create a vote account and start getting delegations in epoch `E - 1`, then those stakes aren't active until the start of epoch `E`, and they should receive rewards after epoch `E`.

Is the idea to use the commission as of the vote account's creation during `E - 1` when calculating payouts at the end of epoch `E`?

Reply:
> If I create a vote account and start getting delegations in epoch `E - 1`, then those stakes aren't active until the start of epoch `E`, and they should receive rewards after epoch `E`.

Actually at earliest they will start receiving rewards for epoch `E + 1` (distributed in epoch `E + 2`) because that's the earliest epoch in which your vote account could be in the leader schedule. We use the stake distribution at the beginning of epoch `E` to determine the leader schedule for `E + 1` and epoch `E` is the earliest epoch in which your vote account would be staked.

Reply:
I moved the commission rate in basis points change to a new SIMD and reverted this SIMD back to dealing with delayed commission rate updates. New SIMD is here: https://github.com/solana-foundation/solana-improvement-documents/pull/291